src/api-client.ts

```diff
@@ -25,9 +26,11 @@ export class RevAiApiClient {
     /**
      * @param accessToken Access token used to validate API requests
      * @param version (optional) version of the API to be used
+     * @param baseUrl (optional) base url of the API to be used.
+     *     Note that non-US global deployments have a different base url
      */
-    constructor (accessToken: string, version = 'v1') {
-        this.apiHandler = new ApiRequestHandler(`https://api.rev.ai/speechtotext/${version}/`, accessToken);
+    constructor (accessToken: string, version = 'v1', baseUrl = BaseUrls.US) {
```

Comment:
my concern with this approach is that you can't have (easily) named parameters, so to specify the `baseUrl` you have to specify the version.

```
new Client(accessToken = '', baseUrl = '');
```

I propose to introduce a new (additional) constructor e.g.
```
interface SomeMagicApiConfig {
  accessToken: string;
  version?: string;
  baseUrl?: string;
}

constructor(SomeMagicApiConfig  config) {
  const mergedConfig = Object.assign(this.defaultConfig, config);
}

const client = new Client({ accessToken: '', baseUrl: '' });
```

in this case users can specify whatever they need without the additional headache.

cc: @k-weng  @amikofalvy

Reply:
I think this would be a huge breaking change

Reply:
this goes in addition to the current ctor so there is no breaking change.
we will use the new ctor in the old one, but any new parameters will be added to the new ctor config only